addons/rating_project/models/project.py

```diff
@@ -21,26 +23,70 @@ class Task(models.Model):
     _name = 'project.task'
     _inherit = ['project.task', 'rating.mixin']
 
+    rating_latest = fields.Float(string="Latest Rating", related="rating_ids.rating", group_operator="avg", store=True)
+    rating_feedback = fields.Text(string="Rating Feedback", related="rating_ids.feedback", store=True)
+    rating_text = fields.Text(string="Rating Text", compute="_get_rating_text")
+
     @api.multi
     def write(self, values):
-        if 'stage_id' in values and values.get('stage_id'):
-            template = self.env['project.task.type'].browse(values.get('stage_id')).rating_template_id
+        result = super(Task, self).write(values)
+        if self.project_id.rating_status == 'stage' and 'stage_id' in values and values.get('stage_id'):
+            self._send_task_rating_mail()
+        return result
+
+    @api.multi
+    def _send_task_rating_mail(self):
+        for task in self:
+            template = task.stage_id.rating_template_id
             if template:
-                rated_partner_id = self.user_id.partner_id
-                partner_id = self.partner_id
-                if partner_id and rated_partner_id:
-                    self.rating_send_request(template, partner_id, rated_partner_id)
-        return super(Task, self).write(values)
+                partner = task.partner_id or None
+                if not partner:
+                    partner = task.sale_line_id.order_id.partner_id or None
+                if not partner:
+                    partner = task.project_id.partner_id or None
+                rated_partner_id = task.user_id.partner_id
+                if partner and rated_partner_id:
+                    task.rating_send_request(template, partner, rated_partner_id)
+
+    @api.one
+    def _get_rating_text(self):
+        if (self.project_id.rating_status == 'no') or (not self.rating_ids):
+            self.rating_text = False
+            return True
+        self.rating_text = {0: _('Not happy'), 5: _('Average'), 10: _('Happy')}.get(self.rating_latest, _('Unknown rating'))
 
 
 class Project(models.Model):
 
     _inherit = "project.project"
 
+    # This method should be called once a day by the scheduler
+    @api.model
+    def _send_rating_all(self):
+        periods = ['daily']
+        if datetime.today().day in (1, 15):
+            periods.append('bimonthly')
+        if datetime.today().day == 1:
+            periods.append('monthly')
+            if datetime.today().month in (1, 4, 7, 10):
+                periods.append('quarterly')
+            if datetime.today().month in (1,):
+                periods.append('yearly')
+        if datetime.today().weekday() == 2:
+            periods.append('weekly')
+        project_ids = self.search([('rating_status', '=', 'periodic'), ('rating_status_period', 'in', periods)])
+        for project in project_ids:
+            project._send_rating_mail()
+
+    def _send_rating_mail(self):
+        self.task_ids._send_task_rating_mail()
```

Comment:
since `_send_task_rating_mail` seems to correctly handle n-ary recordsets, `self.mapped('task_ids')._send_task_rating_mail()` would allow `project_ids._send_rating_mail()` without having to iterate manually I think.